src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/BindingInfo.cs

```diff
@@ -134,6 +134,44 @@ public static BindingInfo GetBindingInfo(IEnumerable<object> attributes)
             return isBindingInfoPresent ? bindingInfo : null;
         }
 
+        public static BindingInfo GetBindingInfo(ModelMetadata modelMetadata)
```

Comment:
My quick read of `ModelBinderFactory` was mistaken: That class does use `ParameterDescriptor.BindingInfo` a bit. And changes made to `ParameterModel`s and `ParameterDescriptor`s _do_ matter. The issue is just that the `BindingInfo` is incomplete. See https://github.com/aspnet/Mvc/blob/a72b56f49e3252376295a35e15523ed3711d80e0/src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBinderFactory.cs#L240-L247

Since `factoryContext.BindingInfo` is usually a copy of `ParameterDescriptor.BindingInfo`, recommend passing an `initialBindingInfo` into this method.